proposals/infra/INF-0004-register-types-and-diagnostics.md

```diff
@@ -45,58 +48,56 @@ The design below aims to specify the answer to these questions.
 Firstly, the most common case is when a resource type that is not a 
 user-defined type is bound as a resource. Any resource will have an 
 associated ResourceAttr attribute, from which we can determine the 
-resource class and resource kind. These two inputs are sufficient to
-determine the expected register type, and any other register type
-will result in an error. The recommended register type will be
-suggested in the diagnostic. The table below specifies what register 
-type will be expected for any resource type that falls under the
-specified resource class. 
+resource class The resource class data is sufficient to determine 
+the expected register type, and any other register type will result 
+in an error. The recommended register type will be suggested in the
+diagnostic. The table below specifies what register type will be 
+expected for any resource type that falls under the specified 
+resource class. 
 
 | Resource Class | Register Type | Diagnostic |
 |-|-|-|
-| Sampler | s | "Object type '%0' with resource class "Sampler" expects register type 's', but register type '%1' was given." |
-| SRV | t | "Object type '%0' with resource class "SRV" expects register type 't', but register type '%1' was given." |
-| UAV | u | "Object type '%0' with resource class "UAV" expects register type 'u', but register type '%1' was given." |
-| CBuffer | b |  "Object type '%0' with resource class "CBuffer" expects register type 'b', but register type '%1' was given." |
+| Sampler | s | "%select{SRV|UAV|CBV|Sampler}2 type '%0' requires register type '%select{t|u|b|s}2', but register type '%1' was used." |
+| SRV | t | "%select{SRV|UAV|CBV|Sampler}2 type '%0' requires register type '%select{t|u|b|s}2', but register type '%1' was used." |
+| UAV | u | "%select{SRV|UAV|CBV|Sampler}2 type '%0' requires register type '%select{t|u|b|s}2', but register type '%1' was used." |
+| CBuffer | b | "%select{SRV|UAV|CBV|Sampler}2 type '%0' requires register type '%select{t|u|b|s}2', but register type '%1' was used." |

```

Comment:
The diagnostic string in the table isn't useful now, since it's identical for each row.  It would be good to provide the diagnostic string in one place (before the table).  Then this is just a table of Resource Class to Register Type.